uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/absence-tab/components/contract-entitlements_test.js

```diff
@@ -0,0 +1,109 @@
+/* eslint-env amd, jasmine */
+
+define([
+  'common/lodash',
+  'common/moment',
+  'common/angular',
+  'common/angularMocks',
+  'leave-absences/shared/config',
+  'leave-absences/absence-tab/app',
+  'common/mocks/services/api/contract-mock',
+  'mocks/apis/absence-type-api-mock',
+  'leave-absences/shared/modules/shared-settings'
+], function (_, moment, angular) {
+  'use strict';
+
+  describe('contactEntitlements', function () {
+    var $compile, $log, $rootScope, $q, component, controller, Contract,
+      ContractAPIMock, AbsenceType, AbsenceTypeAPIMock;
+
+    beforeEach(module('leave-absences.templates', 'absence-tab',
+                      'common.mocks', 'leave-absences.mocks'
+    ));
+
+    beforeEach(inject(['api.contract.mock', 'AbsenceTypeAPIMock', function (_ContractAPIMock_, _AbsenceTypeAPIMock_) {
+      ContractAPIMock = _ContractAPIMock_;
+      AbsenceTypeAPIMock = _AbsenceTypeAPIMock_;
+    }]));
+
+    beforeEach(inject(function (_$compile_, _$log_, _$rootScope_, _$q_, _Contract_, _AbsenceType_) {
+      $compile = _$compile_;
+      $log = _$log_;
+      $rootScope = _$rootScope_;
+      $q = _$q_;
+
+      Contract = _Contract_;
+      AbsenceType = _AbsenceType_;
+
+      spyOn($log, 'debug');
+      spyOn(AbsenceType, 'all').and.callFake(function () {
+        return $q.resolve(AbsenceTypeAPIMock.all());
+      });
+      spyOn(Contract, 'all').and.callFake(function () {
+        return $q.resolve(ContractAPIMock.all());
+      });
+
+      compileComponent();
+    }));
+
+    it('is initialized', function () {
+      expect($log.debug).toHaveBeenCalled();
+    });
+
+    it('has a contact to load for', function () {
+      expect(controller.contactId).toBeDefined();
+    });
+
+    it('has loaded contracts', function () {
+      expect(controller.contracts).not.toBeNull();
+      expect(controller.contracts).toEqual(jasmine.any(Array));
+    });
+
+    describe('contract', function () {
+      var contract;
+
+      beforeEach(function () {
+        contract = controller.contracts[0];
+      });
+
+      it('has position', function () {
```

Comment:
Dont just test if it is defined, we should check if 
`contract.position = mockedData.details.position`
Same goes of other keys too.